internal/handlers/pg/pjson/schema.go

```diff
@@ -125,3 +121,145 @@ func (s *schema) Unmarshal(b []byte) error {
 
 	return nil
 }
+
+// makeSchema makes schema for the given document based on its data.
+func makeSchema(td *types.Document) ([]byte, error) {
```

Comment:
Maybe we can call it `marshalSchema` as it converts the document to a slice of bytes?

Reply:
Suggested a new name (and docs)